Add reproducibility instructions (#271)
AlexsLemonade/scpca-paper-figures: reproduce-figures/README.md

This directory contains a script and certain data files necessary for reproducing manuscript figures, tables, and analyses.
It contains the following contents:

* The script `prepare-scpca-portal-data.R` prepares data obtained from the ScPCA Portal and AWS S3 needed to reproduce figures, tables, and the bulk RNA-Seq analysis.
Please see instructions below for usage.

>   * This script uses functions provided in `utils.R`

Reviewer: ```suggestion
  * **Caution: Running this script requires at least 170 GB of free storage space.**
  * This script uses functions provided in `utils.R`
```

  * This script also uses the `./scratch` directory during file processing

> * TSV files `FigS1B-gene-expression-data.tsv` and `FigS1C_FigS1D-cell-metrics-data.tsv` are used as input to the script `../scripts/FigS1B-D_method-metrics-comparison.R` which creates Figures S1-D.

Reviewer: ```suggestion
* TSV files, `FigS1B-gene-expression-data.tsv` and `FigS1C_FigS1D-cell-metrics-data.tsv`, which are used as input to the script `../scripts/FigS1B-D_method-metrics-comparison.R` to create Figures S1-D.
```
I think this just clarifies that these files live here but are used as input to another script.

  * Note that this script expects these TSV files to be present in this directory; please do not move them.

> **Caution: This process script requires at least 170 GB free storage space.**

Reviewer: ```suggestion
```
This feels like it's out of place here, so I'm suggesting moving it to the above bullets.

## Instructions

To reproduce figures and analyses in this repository, you will need to organize and prepare all data needed for input.
Follow the steps below to do so:

### Step 1: Set up the `renv` environment

This repository uses [`renv`](https://rstudio.github.io/renv/) to manage R packages and dependencies.
To set up the `renv` environment, open the the `scpca-paper-figures.Rproj` project in RStudio and run `renv::restore()` to install all packages in the provided `renv.lock` file.
Note that this project uses R version `4.4.0`.

Please refer to the [Introduction to `renv` vignette](https://rstudio.github.io/renv/articles/renv.html) for more information about getting started with `renv`.

### Step 2: Download files from the ScPCA Portal

You will need to download the following files from the ScPCA Portal <https://scpca.alexslemonade.org/>:

1. All ScPCA projects listed in the [Project Whitelist file](../sample-info/project-whitelist.txt)

>   * On each project page (<https://scpca.alexslemonade.org/projects/SCPCPXXXXXX>), download each project with the following options selected:

Reviewer: I think I might describe how to construct the URL to each project rather than providing a URL that doesn't work. Maybe like this: 

```suggestion
  * Use the project identifiers listed in the project whitelist file to navigate to each project page. 
For example, to navigate to `SCPCP000001`, use the URL: <https://scpca.alexslemonade.org/projects/SCPCP000001>. 
* Download each project using the `Download Project` button with the following options selected: 
```

    * Ensure the selected modality is `Single-cell`
    * Ensure the selected Data Format is `SingleCellExperiment (r)`
    * Do _not_ click to merge all objects into the same sample
    * If the project contains multiplexed samples, you can exclude those samples (but it will not affect the code if they are included)
  * These projects will download as zip files; you should leave them in this compressed format, and _do not rename them_.

2. The portal-wide metadata TSV file
  * You can download this single TSV file using the "Get All Sample Metadata" button on the top-right of the portal homepage.

### Step 3: Organize your downloaded files

> Place all downloaded ScPCA project files are in the same directory.

Reviewer: ```suggestion
Place all downloaded ScPCA project files in the same directory.
```

Optionally, you can also store the portal metadata TSV in this directory, but it is not necessary.

### Step 4: Run the `prepare-scpca-portal-data.R` script

> Now, you can run the `prepare-scpca-portal-data.R` script specifying the following input arguments:

Reviewer: I might add the warning here about how much space this script will use?

Author: added!

* `--portal_projects_dir`: Path to the directory containing the ScPCA project zip files
* `--portal_metadata_path`: Path to the portal-wide metadata TSV file

```sh
  Rscript prepare-scpca-portal-data.R \
    --portal_projects_dir <path to directory with all ScPCA project zip files> \
    --portal_metadata_path <path to portal-wide metadata TSV>
```

This script will run for roughly 90 minutes, and it will create three directories within this repository with files organized as code expects:

> * `../s3_files`: This directory will contain ScPCA data, metadata, and reference files needed to reproduce figures, tables, and the bulk expression analysis

Reviewer: What files are in`s3_files` that are used for bulk analysis?

Author: Consensus cell types are used to make the genesets we use in the over-representation analysis.
That said, these are all in the repository - https://github.com/AlexsLemonade/scpca-paper-figures/tree/main/analysis/pseudobulk-bulk-prediction/data
We could make this an optional step in the bulk pipeline if those exist, but I don't think we need to.

Reviewer: Ah yes! I was just curious since looking at the folder nothing was obvious to me so I just wanted to double check your statement was correct. No need to make any additional code changes!

This directory contains a script and certain data files necessary for reproducing manuscript figures, tables, and analyses.
It contains the following contents:

> * The script `prepare-scpca-portal-data.R` prepares data obtained from the ScPCA Portal and AWS S3 needed to reproduce figures, tables, and the bulk RNA-Seq analysis.

Reviewer: ```suggestion
* The script `prepare-scpca-portal-data.R` which prepares data obtained from the ScPCA Portal and AWS S3 needed to reproduce figures, tables, and the bulk RNA-Seq analysis.
```
Matching the formatting of the second bullet point.
